iped-parsers/iped-parsers-impl/src/main/java/gpinf/emule/PreferencesDatParser.java

```diff
@@ -34,53 +34,53 @@
     }
 
     @Override
-    public void parse(InputStream is, ContentHandler handler, Metadata metadata, ParseContext context)
+    public void parse(InputStream stream, ContentHandler handler, Metadata metadata, ParseContext context)
             throws IOException, SAXException, TikaException {
         // TODO Auto-generated method stub
-        IItemBase item = context.get(IItemBase.class);
-        try (iped3.io.SeekableInputStream stream = item.getStream()) {
-            XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
-            xhtml.startDocument();
-            int version = -1;
-            byte b[] = new byte[16];
-            if (stream.size() > 16) {
-                version = stream.read();
-                for (int i = 0; i < b.length;) {
-                    i += stream.read(b, i, b.length - i);
-                }
 
-                xhtml.startElement("table", "border", "1");
-                xhtml.startElement("tr");
+        XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
+        xhtml.startDocument();
+        int version = -1;
+        byte b[] = new byte[16];
 
-                xhtml.startElement("td");
-                xhtml.characters(Messages.getString("PreferencesDat.Version"));
-                xhtml.endElement("td");
-                xhtml.startElement("td");
-                xhtml.characters(Integer.toString(version));
-                xhtml.endElement("td");
+        version = stream.read();
+        for (int i = 0; i < b.length;) {
+            int readbytes = stream.read(b, i, b.length - i);
+            if (readbytes > 0) {
```

Comment:
I think this should be changed to
```
if (readbytes >= 0) {
    i += readbytes;
} else...
```
Everything else seems good to me, thank you!